proposals/Classes.md

```diff
@@ -0,0 +1,309 @@
+## Classes
+
+Classes will provide one of the basic type building blocks in the project.
+
+### Basic Definition
+
+A `class` is a type which provides a container for holding data (variables/fields) and operational
+members (methods, properties, indexers, operators, static functions, etc).
+
+### Defining a class
+
+A class is defined with the `class` keyword, along with its *members*.
+
+```
+class SomeClass {
+   var<int> someInt1 = 2
+   var<int> someInt2
+   var<int?> someInt3
+   
+   constructor WithInts(int a, int b, int? c) {
+	 this.someInt1 = a
+	 this.someInt2 = b
+	 this.someInt3 = c
+   }
+
+   property<int> SumOfInts {
+	 get {
+	   return someInt1 + someInt2 + (someInt3 ?? 0)
+	 }
+   }
+} 
+```
+
+### Supported class members
+
+The full list to be determined, but initially:
+
+- constructor
+- variable (field)
+- functions
+- properties
+- operators
+
+
+### `constructor`: Constructing classes
+
+A class is constructed by invoking a `constructor` defined on the class: 
+
+```
+SomeClass.SomeConstructor(firstarg, secondarg)
+```
+
+If a class has no constructors defined, a *default constructor* named `new` is defined. This constructor
```

Comment:
I think module private is the right accessibility for this, but comments are welcome